client/app/scripts/superdesk-authoring/authoring.js

```diff
@@ -1382,11 +1397,7 @@
                 };
 
                 scope.$watch('options', function() {
-                    fetchTemplates(scope.options.page, scope.options.keyword)
-                    .then(function(result) {
-                        scope.maxPage = Math.ceil(result._meta.total / PAGE_SIZE);
-                        scope.templates = result;
-                    });
+                    fetchTemplates();
                 }, true);
```

Comment:
I agree, prefer using `$watchCollection()` instead of `$watch(.., .., true)`, when possible. The latter often does too much work, unnecessarily slowing down the `$digest` loop.